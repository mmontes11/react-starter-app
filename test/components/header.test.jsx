import React, { Fragment } from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";
import Header from "components/header";

const Dummy = () => <div>Dummy</div>;

describe("components/header", () => {
  it("renders a header in / path", () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Fragment>
          <Header />
          <Route path="/game" component={Dummy} />
          <Route path="/about" component={Dummy} />
        </Fragment>
      </MemoryRouter>,
    );

    expect(wrapper.find(Dummy)).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a header in /game path", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/game"]} keyLength={0}>
        <Fragment>
          <Header />
          <Route path="/game" component={Dummy} />
          <Route path="/about" component={Dummy} />
        </Fragment>
      </MemoryRouter>,
    );

    expect(wrapper.find('[href="/game"]').hasClass("header-link-active")).toBeTruthy();
    expect(wrapper.find('[href="/about"]').hasClass("header-link-active")).toBeFalsy();
    expect(wrapper.find(Dummy)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a header in /about path", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/about"]} keyLength={0}>
        <Fragment>
          <Header />
          <Route path="/game" component={Dummy} />
          <Route path="/about" component={Dummy} />
        </Fragment>
      </MemoryRouter>,
    );

    expect(wrapper.find('[href="/game"]').hasClass("header-link-active")).toBeFalsy();
    expect(wrapper.find('[href="/about"]').hasClass("header-link-active")).toBeTruthy();
    expect(wrapper.find(Dummy)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
