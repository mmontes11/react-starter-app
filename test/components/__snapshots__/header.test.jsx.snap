// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/header renders a header in / path 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <React.Fragment>
    <Header />
    <Route
      component={[Function]}
      path="/game"
    />
    <Route
      component={[Function]}
      path="/about"
    />
  </React.Fragment>
</Router>
`;

exports[`components/header renders a header in /about path 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/about",
    ]
  }
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/about",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/about",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <header
        className="header"
      >
        <img
          alt="logo"
          className="header-logo"
          src="test-file-stub"
        />
        <h1
          className="header-title"
        >
          React starter app
        </h1>
        <div
          className="header-links"
        >
          <NavLink
            activeClassName="header-link-active"
            aria-current="page"
            className="header-link"
            to="/game"
          >
            <Route
              path="\\\\/game"
            >
              <Link
                aria-current={null}
                className="header-link"
                replace={false}
                to="/game"
              >
                <a
                  aria-current={null}
                  className="header-link"
                  href="/game"
                  onClick={[Function]}
                >
                  Game
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            activeClassName="header-link-active"
            aria-current="page"
            className="header-link"
            to="/about"
          >
            <Route
              path="\\\\/about"
            >
              <Link
                aria-current="page"
                className="header-link header-link-active"
                replace={false}
                style={Object {}}
                to="/about"
              >
                <a
                  aria-current="page"
                  className="header-link header-link-active"
                  href="/about"
                  onClick={[Function]}
                  style={Object {}}
                >
                  About
                </a>
              </Link>
            </Route>
          </NavLink>
        </div>
      </header>
    </Header>
    <Route
      component={[Function]}
      path="/game"
    />
    <Route
      component={[Function]}
      path="/about"
    >
      <Dummy
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/about",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/about",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/about",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/about",
            "url": "/about",
          }
        }
      >
        <div>
          Dummy
        </div>
      </Dummy>
    </Route>
  </Router>
</MemoryRouter>
`;

exports[`components/header renders a header in /game path 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/game",
    ]
  }
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/game",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/game",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <header
        className="header"
      >
        <img
          alt="logo"
          className="header-logo"
          src="test-file-stub"
        />
        <h1
          className="header-title"
        >
          React starter app
        </h1>
        <div
          className="header-links"
        >
          <NavLink
            activeClassName="header-link-active"
            aria-current="page"
            className="header-link"
            to="/game"
          >
            <Route
              path="\\\\/game"
            >
              <Link
                aria-current="page"
                className="header-link header-link-active"
                replace={false}
                style={Object {}}
                to="/game"
              >
                <a
                  aria-current="page"
                  className="header-link header-link-active"
                  href="/game"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Game
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            activeClassName="header-link-active"
            aria-current="page"
            className="header-link"
            to="/about"
          >
            <Route
              path="\\\\/about"
            >
              <Link
                aria-current={null}
                className="header-link"
                replace={false}
                to="/about"
              >
                <a
                  aria-current={null}
                  className="header-link"
                  href="/about"
                  onClick={[Function]}
                >
                  About
                </a>
              </Link>
            </Route>
          </NavLink>
        </div>
      </header>
    </Header>
    <Route
      component={[Function]}
      path="/game"
    >
      <Dummy
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/game",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/game",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/game",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/game",
            "url": "/game",
          }
        }
      >
        <div>
          Dummy
        </div>
      </Dummy>
    </Route>
    <Route
      component={[Function]}
      path="/about"
    />
  </Router>
</MemoryRouter>
`;
